rpc/http.go

```diff
@@ -324,6 +324,7 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	connInfo.HTTP.Host = r.Host
 	connInfo.HTTP.Origin = r.Header.Get("Origin")
 	connInfo.HTTP.UserAgent = r.Header.Get("User-Agent")
+	connInfo.HTTP.Header = r.Header.Clone()
```

Comment:
We could consider not cloning since this is on the hot path of requests, having it contractual that it should not be mutated

Reply:
I'm also not sure. Since the `PeerInfo` doesn't outlive the http request, you're probably right.

Reply:
I'm strongly in favor of not cloning; since it's in the hot-path, for *every* request, and includes all common headers, it is copying a lot of data over time. Including on every RPC interaction on a provider, and every engine API call.

Reply:
+1.

In order to validate signature, the request body must also be available to the endpoint so I updated to also include the `Body` in the PeerInfo.

Alternatively we could enshrine authentication apart of `validateRequest` where the rpc server can be configured to specify the name of the authentication header to look out for. wdyt? @protolambda @sebastianst